reference/snapshot.md

```diff
@@ -0,0 +1,269 @@
+
+Use cases for volume snapshots include:
+
+- Efficient replication for backups.
+- Utilization of clones for troubleshooting.
+- Development against a read-only copy of data.
+
+Volume snapshots allow the creation of read-only incremental copies of volumes, enabling you to maintain a history of your data. These volume snapshots possess the following characteristics:
+- **Consistency**: The data stored within a snapshot remains consistent across all replicas of the volume, whether local or remote.
+- **Immutability**: Once a snapshot is successfully created, the data contained within it cannot be modified.
+
+Currently, Mayastor supports the following operations related to volume snapshots:
+
+1. Creating a snapshot for a PVC
+2. Listing available snapshots for a PVC
+3. Deleting a snapshot for a PVC
+
+:::note
+The current version does not support creation of a clone or a new volume from a snapshot.
+:::
+
+------------------
+
+
+## Prerequisites
+
+
+1. Deploy and configure Mayastor by following the steps given [here](quickstart/deploy-mayastor.md) and create disk pools. 
+2. Create a Mayastor StorageClass with single replica.
+
+:::note
+Currently Mayastor only supports snapshots for volumes with a single replica. Snapshot support for volumes with more than one replica will be available in the future versions.
+:::
+
+{% tabs %}
+{% tab title="Command (single replica)" %}
+```text
+cat <<EOF | kubectl create -f -
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: mayastor-1
+parameters:
+  ioTimeout: "30"
+  protocol: nvmf
+  repl: "1"
+provisioner: io.openebs.csi-mayastor
+EOF
+```
+{% endtab %}
+
+{% tabs %}
+{% tab title="YAML (single replica)" %}
+```text
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: mayastor-1
+parameters:
+  ioTimeout: "30"
+  protocol: nvmf
+  repl: "1"
+provisioner: io.openebs.csi-mayastor
+```
+{% endtab %}
+{% endtabs %}
+
+3. Create a PVC using [these](quickstart/deploy-a-test-application.md) steps and check if the status of the PVC is **Bound**.
+
+{% tabs %}
+{% tab title="Command" %}
+```text
+kubectl get pvc
+```
+{% endtab %}
+{% tab title="Example Output" %}
+```text
+NAME                STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS     AGE
+ms-volume-claim     Bound    pvc-fe1a5a16-ef70-4775-9eac-2f9c67b3cd5b   1Gi        RWO            mayastor-1       15s
+```
+{% endtab %}
+{% endtabs %}
+
+> Copy the PVC name, for example, `ms-volume-claim`.
+
+4. (Optional) Create an application by following [these](quickstart/deploy-a-test-application.md) steps. 
+ 
+
+----------
+ 
+## Create a Snapshot
+
+You can create a snapshot **with or without an application using a PVC**. Follow the steps below to create a volume snapshot:
```

Comment:
@tiagolobocastro @Abhinandan-Purkait "with or without an application using a PVC" is this needed?